rero_ils/modules/documents/commons/identifiers.py

```diff
@@ -0,0 +1,281 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2022 RERO
+# Copyright (C) 2022 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Common classes used for documents - identifiers.
+
+Identifiers class represent all possible identifiers data from a document
+resource : ISBN, ISSN, EAN, GTIN, DOI, ...
+
+As identifiers are very versatile, we can build a single class to validate or
+to normalize them. That is why we will use a factory class to build
+any identifier from data. This factory will analyze data to build the best
+possible specialized identifier class.
+
+HOW TO USE :
+
+   >> from rero_ils.modules.documents.commons import *
+   >> data = {
+   >>    'type': 'bf:Isbn',
+   >>    'value': '9782844267788',
+   >>    'qualifier: 'tome 2'
+   >> }
+   >> identifier = IdentifierFactory.createIdentifier(data)
+   >> print(s.render(render_class=QualifierIdentifierRenderer())
+
+To render an identifier in a specialized way, a `render class` can be used to
+render this identifier in the expected way. In the previous example, we will
+use the `QualifierIdentifierRenderer` ; the result will be
+"9782844267788, tome 2"
+
+"""
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import Optional, TypeVar
+
+from isbnlib import NotValidISBNError, canonical, ean13, is_isbn10, mask, \
+    notisbn, to_isbn10, to_isbn13
+
+
+class InvalidIdentifierException(Exception):
+    """Exception if an identifier is invalid."""
+
+
+class IdentifierType:
+    """Type of identifier."""
+
+    AUDIO_ISSUE_NUMBER = 'bf:AudioIssueNumber'
+    DOI = 'bf:Doi'
+    EAN = 'bf:Ean'
+    GTIN_14 = 'bf:Gtin14Number'
+    IDENTIFIER = 'bf:Identifier'
+    ISAN = 'bf:Isan'
+    ISBN = 'bf:Isbn'
+    ISMN = 'bf:Ismn'
+    ISRC = 'bf:Isrc'
+    ISSN = 'bf:Issn'
+    L_ISSN = 'bf:IssnL'
+    LCCN = 'bf:Lccn'
+    LOCAL = 'bf:Local'
+    MATRIX_NUMBER = 'bf:MatrixNumber'
+    MUSIC_DISTRIBUTOR_NUMBER = 'bf:MusicDistributorNumber'
+    MUSIC_PLATE = 'bf:MusicPlate'
+    MUSIC_PUBLISHER_NUMBER = 'bf:MusicPublisherNumber'
+    PUBLISHER_NUMBER = 'bf:PublisherNumber'
+    UPC = 'bf:Upc'
+    URN = 'bf:Urn'
+    VIDEO_RECORDING_NUMBER = 'bf:VideoRecordingNumber'
+    URI = 'uri'
+
+
+class IdentifierStatus:
```

Comment:
I do not understand what is the purpose of this. You can perhaps use the status of the invenio persistent identifiers:
https://github.com/inveniosoftware/invenio-pidstore/blob/e58e0987e7bcc8d8b021f44a88b401b89b8ae8c8/invenio_pidstore/models.py#L35